integtest/src/main/java/org/corfudb/universe/cluster/docker/DockerCluster.java

```diff
@@ -0,0 +1,136 @@
+
+/**
+ * Represents Docker implementation of a {@link Cluster}.
+ */
+@Slf4j
+@Builder
+@AllArgsConstructor
+public class DockerCluster implements Cluster {
+
+    @Getter
+    private final ClusterParams clusterParams;
+    private final DockerClient docker;
+    private final DockerNetwork network = new DockerNetwork();
+    @Default
+    private final ImmutableList<Service> services = ImmutableList.of();
+
+    /**
+     * Deploy a cluster according to provided cluster parameter, docker client, docker network, and other cluster
+     * components.
+     * The instances of this class are immutable. In other word, when the state of an instance is changed a new
+     * immutable instance is provided.
+     *
+     * @return New immutable instance of docker cluster would be returned.
+     * @throws ClusterException this exception will be thrown if deploying a cluster is not successful
+     */
+    @Override
+    public DockerCluster deploy() throws ClusterException {
+        network.setup();
+
+        List<Service> servicesSnapshot = createAndDeployServices();
+
+        return DockerCluster.builder()
+                .clusterParams(clusterParams)
+                .docker(docker)
+                .services(ImmutableList.copyOf(servicesSnapshot))
+                .build();
+    }
+
+    @Override
+    public void shutdown() throws ClusterException {
+        services.forEach(service -> service.stop(clusterParams.getTimeout()));
+        network.shutdown();
+    }
+
+    @Override
+    public ImmutableList<Service> getServices() {
+        return services;
+    }
+
+    private List<Service> createAndDeployServices() {
+        List<Service> servicesSnapshot = new ArrayList<>();
+
+        for (String serviceName : clusterParams.getServices().keySet()) {
+            DockerService service = DockerService.builder()
+                    .clusterParams(clusterParams)
+                    .params(clusterParams.getService(serviceName))
+                    .docker(docker)
+                    .build();
+
+            service = service.deploy();
+
+            servicesSnapshot.add(service);
+        }
+
+        return servicesSnapshot;
+    }
+
+    private class DockerNetwork {
+        private final Logger log = LoggerFactory.getLogger(DockerNetwork.class);
+
+        /**
+         * Sets up a docker network.
+         *
+         * @throws ClusterException will be thrown if cannot set up a docker network
+         */
+        void setup() throws ClusterException {
+            log.info("Setup network: {}", clusterParams.getNetworkName());
+            NetworkConfig networkConfig = NetworkConfig.builder()
+                    .checkDuplicate(true)
+                    .attachable(true)
+                    .name(clusterParams.getNetworkName())
+                    .build();
+
+            try {
+                docker.createNetwork(networkConfig);
+            } catch (DockerException | InterruptedException e) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either re-interrupt this method or rethrow the "InterruptedException". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2142)